# LAQN Data Analysis and Cleaning

## 1. Introduction
This notebook performs exploratory data analysis on the LAQN (London Air Quality Network) dataset for research. The goal is to understand the data structure, identify data quality issues, and prepare a cleaned dataset for further analysis.

## 2. Setup and Import Libraries

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
import pathlib
import glob
import sys
import modulefinder as mdfind

# I need to set the path of the datasets here before loading them.

## 3. Load the Dataset : sites_species_london.csv

In [20]:
# ensure notebook runs from project root
os.chdir("/Users/burdzhuchaglayan/Desktop/data science projects/air-pollution-levels")
print("cwd now:", os.getcwd())

# Load the dataset
df = pd.read_csv("data/laqn/sites_species_london.csv", encoding="utf-8")
print("Loaded:", df.shape)

# print statements to verify shape of the df
print(f"Shape: {df.shape[0]} rows, {df.shape[1]} columns")


cwd now: /Users/burdzhuchaglayan/Desktop/data science projects/air-pollution-levels
Loaded: (647, 18)
Shape: 647 rows, 18 columns


## 4. Understanding the Data Structure

### 4.1 First Look at the Data:
- Shape: 647 rows, 18 columns

In [ ]:
#describe the dataset

print('Description of sites_species_london.csv:'(df.describe(include='all')))

#to get detailed information about the dataset.
def info():
    buffer = []
    df.info(buf=buffer)
    s = "\n".join(buffer)
    print(s)
print("Information about the dataset: "(info())

# print first few rows of the dataframe
print(df.head(5))

   @LocalAuthorityCode   @LocalAuthorityName @SiteCode  \
0                   27              Richmond       TD0   
1                   27              Richmond       TD0   
2                   27              Richmond       TD0   
3                   27              Richmond       TD0   
4                    1  Barking and Dagenham       BG3   

                                    @SiteName @SiteType          @DateClosed  \
0  - National Physical Laboratory, Teddington  Suburban  2018-01-01 00:00:00   
1  - National Physical Laboratory, Teddington  Suburban  2018-01-01 00:00:00   
2  - National Physical Laboratory, Teddington  Suburban  2018-01-01 00:00:00   
3  - National Physical Laboratory, Teddington  Suburban  2018-01-01 00:00:00   
4         Barking and Dagenham - North Street  Kerbside  2011-05-25 00:00:00   

           @DateOpened  @Latitude  @Longitude  @LatitudeWGS84  \
0  1996-08-08 00:00:00  51.424304   -0.345715    6.696103e+06   
1  1996-08-08 00:00:00  51.424304   -0.3